src/components/KTableData/KTableData.vue

```diff
@@ -172,91 +170,90 @@ import type {
   PageChangeData,
   PageSizeChangeData,
   TableState,
-  TableStatePayload,
   SwrvStateData,
   TableDataProps,
-  RowExpandPayload,
-  RowActionsTogglePayload,
+  TableColumnKey,
+  TableColumnVisibility,
+  TableColumnWidths,
+  TableDataEmits,
+  TableDataSlots,
 } from '@/types'
 import { EmptyStateIconVariants } from '@/types'
 import { getInitialPageSize, DEFAULT_PAGE_SIZE } from '@/utilities'
 
-const props = withDefaults(defineProps<TableDataProps>(), {
-  resizeColumns: false,
-  tablePreferences: () => ({}),
-  rowHover: true,
-  rowAttrs: () => ({}),
-  rowLink: () => ({} as RowLink),
-  rowBulkActionEnabled: () => true,
-  rowKey: '',
-  cellAttrs: () => ({}),
-  loading: false,
-  emptyStateTitle: 'No Data',
-  emptyStateMessage: 'There is no data to display.',
-  emptyStateActionMessage: '',
-  emptyStateIconVariant: EmptyStateIconVariants.Default,
-  error: false,
-  errorStateTitle: 'An error occurred',
-  errorStateMessage: 'Data cannot be displayed due to an error.',
-  errorStateActionMessage: '',
-  maxHeight: 'none',
-  hidePagination: false,
-  paginationAttributes: () => ({}),
-  rowExpandable: () => false,
-  rowExpanded: () => false,
-  hideHeaders: false,
-  nested: false,
-  hidePaginationWhenOptional: false,
-  hideToolbar: false,
-  /**
-   * KTableData props defaults
-   */
-  headers: () => ([]),
-  fetcherCacheKey: '',
-  cacheIdentifier: '',
-  searchInput: '',
-  initialFetcherParams: () => ({}),
-  clientSort: false,
-  sortable: true,
-})
+type ColumnKey = TableColumnKey<Header>
+type ColumnVisibility = TableColumnVisibility<Header>
+type ColumnWidths = TableColumnWidths<Header>
 
-const slots = useSlots()
+const {
+  resizeColumns,
+  tablePreferences = {},
+  rowHover = true,
+  rowAttrs = () => ({}),
+  rowLink = () => ({} as RowLink),
+  rowBulkActionEnabled = () => true,
+  rowKey: rowKeyProp,
```

Comment:
I'm probably missing something obvious, but where does `rowKeyProp` come from here?

Reply:
There seems to be no need to rename here. Changed.